krakenapi.go

```diff
@@ -601,20 +606,36 @@ func (api *KrakenAPI) queryPrivate(method string, values url.Values, typ interfa
 		"API-Sign": signature,
 	}
 
-	resp, err := api.doRequest(reqURL, values, headers, typ)
+	resp, err := api.doPost(reqURL, values, headers, typ)
 
 	return resp, err
 }
 
-// doRequest executes a HTTP Request to the Kraken API and returns the result
-func (api *KrakenAPI) doRequest(reqURL string, values url.Values, headers map[string]string, typ interface{}) (interface{}, error) {
+func (api *KrakenAPI) doGet(reqURL string, values url.Values, headers map[string]string, typ interface{}) (interface{}, error) {
+	encodedValues := values.Encode()
+	fullURL := reqURL + "?" + encodedValues
+
+	req, err := http.NewRequest("GET", fullURL, nil)
+	if err != nil {
+		return nil, fmt.Errorf("Could not execute request! #1 (%s)", err.Error())
+	}
+
+	return api.doAPIRequest(req, headers, err, typ)
+}
+
+// doPost executes a HTTP Request to the Kraken API and returns the result
+func (api *KrakenAPI) doPost(reqURL string, values url.Values, headers map[string]string, typ interface{}) (interface{}, error) {
 
 	// Create request
 	req, err := http.NewRequest("POST", reqURL, strings.NewReader(values.Encode()))
 	if err != nil {
 		return nil, fmt.Errorf("Could not execute request! #1 (%s)", err.Error())
 	}
 
+	return api.doAPIRequest(req, headers, err, typ)
+}
+
+func (api *KrakenAPI) doAPIRequest(req *http.Request, headers map[string]string, err error, typ interface{}) (interface{}, error) {
```

Comment:
why err in input params here?

Reply:
Thanks for point out. Just pushed a commit to fix it.